docs/repository_management.rst

```diff
@@ -0,0 +1,52 @@
+
+.. _repository_management:
+
+Repository management
+======================
+
+One of the hardest challenges we faced with Cekit is how to manage and define package repositories
+correctly. Our current solution works in following scenarios:
+
+1) Building CentOS or Fedora based images
+2) Building RHEL based images on subscribed hosts
+3) Building RHEL based images on unsubscribed hosts
+
+
+Best Practices
+--------------
+
+To achieve such behavior in Red Hat Middleware Container Images we created following rules and suggestions.
+
+Defining repositories in container images
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You should use :ref:`Plain<repo_plain>` repository definition for every repository as this will work easily on Red Hat subscribed host and will assume everyone can rebuild are RHEL based images.
+
+
+*Example:* Define Software Collections repository
+
+.. code:: yaml
+
+    packages:
+        repositories:
+            - name: SCL
+              id: rhel-server-rhscl-7-rpms
+
+If you have repository defined this way, Cekit will not try to enable it and will expect the repository to be forwarded from your host via Docker subscription plugin. When you want to build our image with CentOS base you need to override this repository. To override a repository definition you need to specify a repository with same ``name``.
```

Comment:
I would add here information that overriding means to provide a definition of the repository that is using any other repository type (besides plain) and link to the definition.

I would also add a second example, for example with path to a repo file that would point to the SCL repository inside of the company. This will make it clear that there are many ways to implement it.